packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -747,13 +748,24 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
 
     // Only check for gestures if tooltip should be visible.
     if (_visible) {
-      result = GestureDetector(
-        behavior: HitTestBehavior.opaque,
-        onLongPress: (_triggerMode == TooltipTriggerMode.longPress) ? _handlePress : null,
-        onTap: (_triggerMode == TooltipTriggerMode.tap) ? _handleTap : null,
-        excludeFromSemantics: true,
-        child: result,
-      );
+      if (_triggerMode == TooltipTriggerMode.tap) {
+        result = GestureDetector(
+          behavior: HitTestBehavior.opaque,
+          onTap: _handleTap,
+          child: result,
+        );
+      }
+      else {
+        // if _triggerMode is not tap exclude GestureDetector from semantics
+        // otherwise in cases the tooltip is used with a button the tap action
+        // on the button will be lost.
```

Comment:
I am not sure if I understand the reasoning. If this does not exclude semantics, we won't need to create a semantics in line 741 right?